<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Enrichment analysis of differential gene expression data • luandre</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Enrichment analysis of differential gene expression data">
<meta property="og:description" content="Package with enrichment analysis tools for bulk RNA-seq data. 
    With quality processed RNA-seq data, one can use LuAndre to filter 
    biological unrelevant gene counts and conduct an differential analysis by 
    fitting a general linear model. Either with a wrapper or step-wise workflow.
    In the end, a summarised excel file and multiple enrichment plots will be
    stored in your working directory.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">luandre</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="luandre">LuAndre<a class="anchor" aria-label="anchor" href="#luandre"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>Package created for Differential Expression Analysis of bulk RNA-seq data.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>LuAndre is not available on CRAN: Install the development version of LuAndre from <a href="https://github.com/ndreey/luandre" class="external-link">GitHub</a> use:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ndreey/luandre"</span><span class="op">)</span></span></code></pre></div>
<p>Manual installment of Bioconductor packages might be needed.</p>
<pre><code>if(BiocManager::texttocome)</code></pre>
</div>
<div class="section level2">
<h2 id="tutorials">Tutorials<a class="anchor" aria-label="anchor" href="#tutorials"></a>
</h2>
<ul>
<li>Check out the <a href="link">LuAndre Manual</a> for more information</li>
</ul>
</div>
<div class="section level2">
<h2 id="enrichment-analysis">Enrichment Analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h2>
<div class="section level3">
<h3 id="option-1-use-the-wrapper">Option 1: Use the wrapper<a class="anchor" aria-label="anchor" href="#option-1-use-the-wrapper"></a>
</h3>
<p>This wrapper orchestrates the necessary functions calls in order</p>
<p><code>{r eval=FALSE} # Run Harmony to integrate the reference cells enrich_analysis("E-MTAB-2523.counts.txt",   # counts table                 "E-MTAB-2523.sample.txt",   # sample table                 db = "GO",                  # "KEGG" or "GO"                 ontology = "BP",            # Biological Process                 threshold = 2)              # Filtering threshold</code> Note that <code>ontology =</code> is only required if <code>db = "GO"</code>is set.</p>
</div>
<div class="section level3">
<h3 id="option-2-step-wise-analysis">Option 2: Step-wise analysis<a class="anchor" aria-label="anchor" href="#option-2-step-wise-analysis"></a>
</h3>
<p>The plots and excel file will be stored in your working directory.</p>
<p>```{r eval=FALSE} # Import data fetched_data &lt;- fetch_data_from_file(“E-MTAB-2523.counts.txt”, “E-MTAB-2523.sample.txt”) # Lets check the dims dim(fetched_data[[1]]) # 19351 genes, 18 samples dim(fetched_data[[2]]) # 18 samples, 3 sample info columns.</p>
</div>
</div>
</div>
<div class="section level1">
<h1 id="lets-filter-the-data-to-remove-low-expression-counts">Lets filter the data to remove low expression counts<a class="anchor" aria-label="anchor" href="#lets-filter-the-data-to-remove-low-expression-counts"></a>
</h1>
<p>filtered_data &lt;- filter(fetched_data, threshold = 3)</p>
</div>
<div class="section level1">
<h1 id="lets-check-how-many-gene-are-kept">Lets check how many gene are kept<a class="anchor" aria-label="anchor" href="#lets-check-how-many-gene-are-kept"></a>
</h1>
<p>dim(filtered_data[[1]]) # 10730 genes</p>
</div>
<div class="section level1">
<h1 id="lets-create-a-digital-gene-expression-list-and-a-design-matrix-using-data3">Lets create a Digital gene expression list and a design matrix using data3<a class="anchor" aria-label="anchor" href="#lets-create-a-digital-gene-expression-list-and-a-design-matrix-using-data3"></a>
</h1>
<p>digital_list &lt;- digital_gene_expression(filtered_data)</p>
</div>
<div class="section level1">
<h1 id="lets-fit-it-to-a-general-linear-model">Lets fit it to a general linear model<a class="anchor" aria-label="anchor" href="#lets-fit-it-to-a-general-linear-model"></a>
</h1>
<p>results &lt;- fit_test_model(digital_list)</p>
</div>
<div class="section level1">
<h1 id="lets-filter-out-degs-that-dont-meet-our-requirments">Lets filter out degs that dont meet our requirments.<a class="anchor" aria-label="anchor" href="#lets-filter-out-degs-that-dont-meet-our-requirments"></a>
</h1>
<p>deg_df &lt;- get_deg(results, lfc = 2, adj_p = 0.01) dim(deg_df) # 180 DEGs!</p>
</div>
<div class="section level1">
<h1 id="lets-store-them-to-an-excel-file">Lets store them to an excel file.<a class="anchor" aria-label="anchor" href="#lets-store-them-to-an-excel-file"></a>
</h1>
<p>save_deg_excel(deg_df3, “E-MTAB-2523”)</p>
</div>
<div class="section level1">
<h1 id="prepare-for-plotting">Prepare for plotting<a class="anchor" aria-label="anchor" href="#prepare-for-plotting"></a>
</h1>
</div>
<div class="section level1">
<h1 id="adds-entrezid-column-to-data-frame">Adds ENTREZID column to data frame<a class="anchor" aria-label="anchor" href="#adds-entrezid-column-to-data-frame"></a>
</h1>
<p>deg_df &lt;- get_Hs_entrez(deg_df)</p>
</div>
<div class="section level1">
<h1 id="enrich-genes-with-kegg-or-go-homo-sapiens">Enrich genes with KEGG or GO (Homo sapiens)<a class="anchor" aria-label="anchor" href="#enrich-genes-with-kegg-or-go-homo-sapiens"></a>
</h1>
<p>enrich_results &lt;- enrich_Hs_genes(deg_df$ENTREZID, db = “GO”, ontology = “BP”)</p>
</div>
<div class="section level1">
<h1 id="plots-for-over-representaiton-data">Plots for over-representaiton data.<a class="anchor" aria-label="anchor" href="#plots-for-over-representaiton-data"></a>
</h1>
<p>enrich_plots(enrich_results, n = 10, font_size = 10, file_id = “E-MTAB-2523”, width = 8, height = 6, dpi = 300) ```</p>
</div>
<div class="section level1">
<h1 id="installation-notes">Installation notes<a class="anchor" aria-label="anchor" href="#installation-notes"></a>
</h1>
<div class="section level2">
<h2 id="system-requirements">System requirements:<a class="anchor" aria-label="anchor" href="#system-requirements"></a>
</h2>
<p>LuAndre has been successfully installed on windows 11 using the devtools package to install from GitHub.</p>
<p>Dependencies:</p>
<ul>
<li>R&gt;=3.6.x</li>
<li>edgeR</li>
<li>limma</li>
<li>clusterProfiler</li>
<li>org.Hs.eg.db</li>
<li>ggplot2</li>
<li>enrichplot</li>
<li>GOSemSim</li>
<li>ggupset</li>
<li>openxlsx</li>
<li>AnnotationDbi</li>
</ul>
</div>
<div class="section level2">
<h2 id="troubleshooting">Troubleshooting:<a class="anchor" aria-label="anchor" href="#troubleshooting"></a>
</h2>
<ul>
<li>You may need to install the latest version of devtools (because of the recent GitHub change from “master” to “main” terminology, which can cause <code>install_github</code> to fail).</li>
</ul>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing luandre</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>André Bourbonnais <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by André Bourbonnais.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
