<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Enrichment analysis of differential gene expression data • luandre</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Enrichment analysis of differential gene expression data">
<meta property="og:description" content="Package with enrichment analysis tools for bulk RNA-seq data. 
    With quality processed RNA-seq data, one can use LuAndre to filter 
    biological unrelevant gene counts and conduct an differential analysis by 
    fitting a general linear model. Either with a wrapper or step-wise workflow.
    In the end, a summarised excel file and multiple enrichment plots will be
    stored in your working directory.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">luandre</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="bi503g-r-package-build-seminar">BI503G R-Package Build Seminar<a class="anchor" aria-label="anchor" href="#bi503g-r-package-build-seminar"></a>
</h1></div>
<!-- README.md is generated from README.Rmd. Please edit that file -->
</div>
<div class="section level1">
<h1 id="luandre">LuAndre<a class="anchor" aria-label="anchor" href="#luandre"></a>
</h1>
<!-- badges: start -->
<!-- badges: end -->
<p>Package created for Differential Expression Analysis of bulk RNA-seq data.</p>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<p>LuAndre is not available on CRAN: Install the development version of LuAndre from <a href="https://github.com/ndreey/luandre" class="external-link">GitHub</a> use:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">if (!require("pacman")) install.packages("pacman")</span></span>
<span><span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">p_load</a></span><span class="op">(</span><span class="st">(devtools, AnnotationDbi, GOSemSim, enrichplot, org.Hs.eg.db, clusterProfiler, limma, edgeR)</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="tutorials">Tutorials<a class="anchor" aria-label="anchor" href="#tutorials"></a>
</h2>
<ul>
<li>Check out the <a href="https://raw.githack.com/ndreey/seminar/main/docs/index.html">LuAndre Manual</a> for more information</li>
</ul>
</div>
<div class="section level2">
<h2 id="enrichment-analysis">Enrichment Analysis<a class="anchor" aria-label="anchor" href="#enrichment-analysis"></a>
</h2>
<div class="section level3">
<h3 id="option-1-use-the-wrapper">Option 1: Use the wrapper<a class="anchor" aria-label="anchor" href="#option-1-use-the-wrapper"></a>
</h3>
<p>This wrapper orchestrates the necessary functions calls in order</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run Harmony to integrate the reference cells</span></span>
<span><span class="fu"><a href="reference/enrich_analysis.html">enrich_analysis</a></span><span class="op">(</span><span class="st">"E-MTAB-2523.counts.txt"</span>,   <span class="co"># counts table</span></span>
<span>                <span class="st">"E-MTAB-2523.sample.txt"</span>,   <span class="co"># sample table</span></span>
<span>                db <span class="op">=</span> <span class="st">"GO"</span>,                  <span class="co"># "KEGG" or "GO"</span></span>
<span>                ontology <span class="op">=</span> <span class="st">"BP"</span>,            <span class="co"># Biological Process</span></span>
<span>                threshold <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>              <span class="co"># Filtering threshold</span></span></code></pre></div>
<p>Note that <code>ontology =</code> is only required if <code>"GO"</code>is set.</p>
</div>
<div class="section level3">
<h3 id="option-2-step-wise-analysis">Option 2: Step-wise analysis<a class="anchor" aria-label="anchor" href="#option-2-step-wise-analysis"></a>
</h3>
<p>The plots and excel file will be stored in your working directory.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Import data</span></span>
<span><span class="va">fetched_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/fetch_data_from_file.html">fetch_data_from_file</a></span><span class="op">(</span><span class="st">"E-MTAB-2523.counts.txt"</span>,</span>
<span>                                   <span class="st">"E-MTAB-2523.sample.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Lets check the dims</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">fetched_data</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>       <span class="co"># 19351 genes, 18 samples</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">fetched_data</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>       <span class="co"># 18 samples, 3 sample info columns.</span></span>
<span></span>
<span><span class="co"># Lets filter the data to remove low expression counts</span></span>
<span><span class="va">filtered_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/filter.html">filter</a></span><span class="op">(</span><span class="va">fetched_data</span>, threshold <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Lets check how many gene are kept</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">filtered_data</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>       <span class="co"># 10730 genes</span></span>
<span></span>
<span><span class="co"># Lets create a Digital gene expression list and a design matrix using data3</span></span>
<span><span class="va">digital_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/digital_gene_expression.html">digital_gene_expression</a></span><span class="op">(</span><span class="va">filtered_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Lets fit it to a general linear model</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/fit_test_model.html">fit_test_model</a></span><span class="op">(</span><span class="va">digital_list</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Lets filter out degs that dont meet our requirments.</span></span>
<span><span class="va">deg_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_deg.html">get_deg</a></span><span class="op">(</span><span class="va">results</span>, lfc <span class="op">=</span> <span class="fl">2</span>, adj_p <span class="op">=</span> <span class="fl">0.01</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">deg_df</span><span class="op">)</span>      <span class="co"># 180 DEGs!</span></span>
<span></span>
<span><span class="co"># Lets store them to an excel file.</span></span>
<span><span class="fu"><a href="reference/save_deg_excel.html">save_deg_excel</a></span><span class="op">(</span><span class="va">deg_df3</span>, <span class="st">"E-MTAB-2523"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Prepare for plotting</span></span>
<span><span class="co"># Adds ENTREZID column to data frame</span></span>
<span><span class="va">deg_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_Hs_entrez.html">get_Hs_entrez</a></span><span class="op">(</span><span class="va">deg_df</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Enrich genes with KEGG or GO (Homo sapiens)</span></span>
<span><span class="va">enrich_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/enrich_Hs_genes.html">enrich_Hs_genes</a></span><span class="op">(</span><span class="va">deg_df</span><span class="op">$</span><span class="va">ENTREZID</span>, db <span class="op">=</span> <span class="st">"GO"</span>, ontology <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plots for over-representaiton data.</span></span>
<span><span class="fu"><a href="reference/enrich_plots.html">enrich_plots</a></span><span class="op">(</span><span class="va">enrich_results</span>, n <span class="op">=</span> <span class="fl">10</span>, font_size <span class="op">=</span> <span class="fl">10</span>, file_id <span class="op">=</span> <span class="st">"E-MTAB-2523"</span>, </span>
<span>             width <span class="op">=</span> <span class="fl">8</span>, height <span class="op">=</span> <span class="fl">6</span>, dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="installation-notes">Installation notes<a class="anchor" aria-label="anchor" href="#installation-notes"></a>
</h1>
<div class="section level2">
<h2 id="system-requirements">System requirements:<a class="anchor" aria-label="anchor" href="#system-requirements"></a>
</h2>
<p>LuAndre has been successfully installed on windows 11 using the devtools package to install from GitHub.</p>
<p>Dependencies:</p>
<ul>
<li>R&gt;=3.6.x</li>
<li>edgeR</li>
<li>limma</li>
<li>clusterProfiler</li>
<li>org.Hs.eg.db</li>
<li>ggplot2</li>
<li>enrichplot</li>
<li>GOSemSim</li>
<li>ggupset</li>
<li>openxlsx</li>
<li>AnnotationDbi</li>
</ul>
</div>
<div class="section level2">
<h2 id="troubleshooting">Troubleshooting:<a class="anchor" aria-label="anchor" href="#troubleshooting"></a>
</h2>
<ul>
<li>You may need to install the latest version of devtools (because of the recent GitHub change from “master” to “main” terminology, which can cause <code>install_github</code> to fail).</li>
</ul>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing luandre</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>André Bourbonnais, Luan Gardenalli <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by André Bourbonnais, Luan Gardenalli</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
